int32 id
int32 sizeOfNodes
int32[] path